lm_eval/models/gpt2.py

```diff
@@ -26,20 +28,27 @@ def __init__(
         assert isinstance(device, str)
         assert isinstance(pretrained, str)
         assert isinstance(batch_size, int)
+        
+        gpus = torch.cuda.device_count()
+        if gpus <= 1:
+            if device:  
+                if device not in ["cuda", "cpu"]:
+                    device = int(device)
+                self._device = torch.device(device)
+                print(f"Using device '{device}'")
+            else:
+                print("Device not specified")
+                print(f"Cuda Available? {torch.cuda.is_available()}")
+                self._device = (
+                    torch.device("cuda")
+                    if torch.cuda.is_available()
+                    else torch.device("cpu")
+                )
+            self._rank = 0
+            self._world_size = 1
```

Comment:
Is there probably a better way to detect if the process was launched with `accelerate launch` or just `python`? I think detecting this would be a more efficient way process both the default process here and the `Accelerator.prepare` process below.

Reply:
I agree that this is a bit messy. I also had that idea, but in looking around, I wasn't able to find a good way to do this...